scanner/engine/evaluate_worker.cpp

```diff
@@ -403,11 +404,33 @@ EvaluateWorker::EvaluateWorker(const EvaluateWorkerArgs& args)
         LOG(ERROR) << "Kernel validate failed: " << args.result.msg();
         THREAD_RETURN_SUCCESS();
       }
+
+      // If we're on worker 0, then fetch resources for the kernel
+      if (worker_id_ == 0) {
+        {
+          Result result;
+          kernel->fetch_resources(&result);
+          args.result.set_msg(result.msg());
+          args.result.set_success(result.success());
+        }
```

Comment:
One possible issue here is that we are sequentially fetching resources. We could possibly parallelize this across multiple threads for each kernel, though that could introduce its own issues. Maybe just leave a note here for now and leave it as it is until it shows up as a problem.

Reply:
Good point. I'll add a note.